src/backend/distributed/executor/multi_real_time_executor.c

```diff
@@ -433,8 +434,29 @@ ManageTaskExecution(Task *task, TaskExecution *taskExecution,
 			 * on this node again.
 			 */
 			int32 connectionId = connectionIdArray[currentIndex];
-			MultiConnection *connection = MultiClientGetConnection(connectionId);
-			bool isCritical = connection->remoteTransaction.transactionCritical;
+			MultiConnection *connection = NULL;
+
+			/* try next worker node next time we look at this task */
+			taskStatusArray[currentIndex] = EXEC_TASK_CONNECT_START;
+			AdjustStateForFailure(taskExecution);
```

Comment:
This was previously only called if `isCritical` was false; I was concerned that the side effects it produces might result in some change in behavior, but since the only new behavior is calling it when `isCritical` is (eventually) true, and when `isCritical` is true we're going to fail-fast anyhow, this is only a no-op as far as behavior is concerned (i.e. the old code path was essentially an optimization of sorts; there is no effect calling it during a critical failure).

Chatted with @marcocitus over Slack to confirm this; otherwise the changes in this function are only changes in ordering.